cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -288,9 +293,10 @@ class device_row_comparator {
  */
 template <typename Comparator, weak_ordering... values>
 struct weak_ordering_comparator_impl {
-  __device__ bool operator()(size_type const lhs, size_type const rhs) const noexcept
+  template <typename... Ts>
+  __device__ bool operator()(Ts&&... args) const noexcept
```

Comment:
We're a bit inconsistent with this, but since we're especially sensitive to trying to boil away all the layers here I would advocate for being more liberal with `constexpr`-ing all the things. 

https://www.youtube.com/watch?v=OcyAmlTZfgg

```suggestion
  __device__ constexpr bool operator()(Ts const... args) const noexcept
```

![image](https://user-images.githubusercontent.com/15221289/168379217-c4abe958-4f0b-43d6-a7d1-3929d398b043.png)